File: src/main/java/com/globaldefensesurveillancetracking/service/EquipmentService.java
package com.globaldefensesurveillancetracking.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.globaldefensesurveillancetracking.domain.Equipment;
import com.globaldefensesurveillancetracking.dto.EquipmentDTO;
import com.globaldefensesurveillancetracking.dto.EquipmentSearchDTO;
import com.globaldefensesurveillancetracking.dto.EquipmentPageDTO;
import com.globaldefensesurveillancetracking.dto.EquipmentConvertCriteriaDTO;
import com.globaldefensesurveillancetracking.service.GenericService;
import com.globaldefensesurveillancetracking.dto.common.RequestDTO;
import com.globaldefensesurveillancetracking.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface EquipmentService extends GenericService<Equipment, Integer> {

	List<Equipment> findAll();

	ResultDTO addEquipment(EquipmentDTO equipmentDTO, RequestDTO requestDTO);

	ResultDTO updateEquipment(EquipmentDTO equipmentDTO, RequestDTO requestDTO);

    Page<Equipment> getAllEquipments(Pageable pageable);

    Page<Equipment> getAllEquipments(Specification<Equipment> spec, Pageable pageable);

	ResponseEntity<EquipmentPageDTO> getEquipments(EquipmentSearchDTO equipmentSearchDTO);
	
	List<EquipmentDTO> convertEquipmentsToEquipmentDTOs(List<Equipment> equipments, EquipmentConvertCriteriaDTO convertCriteria);

	EquipmentDTO getEquipmentDTOById(Integer equipmentId);







}

File: globaldefensesurveillancetracking_ui/src/services/EquipmentService.js
import http from "../http-common";

class EquipmentService {
  getAllEquipments(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/equipment/equipments`, searchDTO);
  }

  get(equipmentId) {
    return this.getRequest(`/equipment/${equipmentId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/equipment?field=${matchData}`, null);
  }

  addEquipment(data) {
    return http.post("/equipment/addEquipment", data);
  }

  update(data) {
  	return http.post("/equipment/updateEquipment", data);
  }
  
  uploadImage(data,equipmentId) {
  	return http.postForm("/equipment/uploadImage/"+equipmentId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new EquipmentService();
